File: src/gui/cryoptix/src/features/api/messages/message-type.ts
import { z } from "zod";

export const MessageType = {
  None: 0,
  MarketDataSnapshot: 1,
  Kline: 2,
  Trade: 3,
  StrategyStarted: 4,
  StrategyUpdated: 5,
} as const;

export const MessageTypeSchema = z.enum(MessageType);

export type MessageType = z.infer<typeof MessageTypeSchema>;

File: src/api/Cryoptix.Observer/Notification/MessageType.cs
namespace Cryoptix.Observer.Notification
{
    /// <summary>
    /// Defines the message type values.
    /// </summary>
    public enum MessageType
    {
        /// <summary>
        /// Specifies the none value.
        /// </summary>
        None,
        /// <summary>
        /// Specifies the strategy started value.
        /// </summary>
        StrategyStarted,
        /// <summary>
        /// Specifies the strategy updated value.
        /// </summary>
        StrategyUpdated,
        /// <summary>
        /// Specifies the market data snapshot value.
        /// </summary>
        MarketDataSnapshot,
        /// <summary>
        /// Specifies the kline value.
        /// </summary>
        Kline,
        /// <summary>
        /// Specifies the trade value.
        /// </summary>
        Trade,
        /// <summary>
        /// Specifies the indicators value.
        /// </summary>
        Indicators,
        /// <summary>
        /// Specifies the signal value.
        /// </summary>
        Signal
    }
}
